Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectFlatPlateAbsorption2.py

```diff
@@ -0,0 +1,436 @@
+            add_sample_log_alg.setProperty("LogValues", log_values)
+            add_sample_log_alg.execute()
+
+        self.setProperty('OutputWorkspace', self._output_ws)
+
+        # Output the Ass workspace if it is wanted, delete if not
+        if self._abs_ws == '':
+            delete_alg.setProperty("Workspace", self._ass_ws)
+            delete_alg.execute()
+            if self._can_ws_name is not None and self._use_can_corrections:
+                delete_alg.setProperty("Workspace", self._acc_ws)
+                delete_alg.execute()
+        else:
+            GroupWorkspaces(InputWorkspaces=group,
+                            OutputWorkspace=self._abs_ws,
+                            EnableLogging=False)
+            self.setProperty('CorrectionsWorkspace', self._abs_ws)
+
+    def _setup(self):
+        """
+        Get algorithm properties.
+        """
+
+        self._sample_ws_name = self.getPropertyValue('SampleWorkspace')
+        self._sample_chemical_formula = self.getPropertyValue('SampleChemicalFormula')
+        self._sample_density_type = self.getPropertyValue('SampleDensityType')
+        self._sample_density = self.getProperty('SampleDensity').value
+        self._sample_height = self.getProperty('SampleHeight').value
+        self._sample_width = self.getProperty('SampleWidth').value
+        self._sample_thickness = self.getProperty('SampleThickness').value
+        self._sample_angle = self.getProperty('SampleAngle').value
+
+        self._can_ws_name = self.getPropertyValue('CanWorkspace')
+        if self._can_ws_name == '':
+            self._can_ws_name = None
+        self._use_can_corrections = self.getProperty('UseCanCorrections').value
+        self._can_chemical_formula = self.getPropertyValue('CanChemicalFormula')
+        self._can_density_type = self.getPropertyValue('CanDensityType')
+        self._can_density = self.getProperty('CanDensity').value
+        self._can_front_thickness = self.getProperty('CanFrontThickness').value
+        self._can_back_thickness = self.getProperty('CanBackThickness').value
+        self._can_scale = self.getProperty('CanScaleFactor').value
+
+        self._emode = 'Indirect'
+        self._efixed = self._getEfixed()
+        self._number_wavelengths = self.getProperty('NumberWavelengths').value
+        self._events = self.getProperty('Events').value
+
+        self._beam_height = self.getProperty('BeamHeight').value
+        self._beam_width = self.getProperty('BeamWidth').value
+
+        self._output_ws = self.getPropertyValue('OutputWorkspace')
+
+        self._abs_ws = self.getPropertyValue('CorrectionsWorkspace')
+        if self._abs_ws == '':
+            self._ass_ws = '__ass'
+            self._acc_ws = '__acc'
+        else:
+            self._ass_ws = self._abs_ws + '_ass'
+            self._acc_ws = self._abs_ws + '_acc'
+
+        # Get beam size defaults
+        # Hard-coded because the ISIS parameters weren't being picked up
```

Comment:
What is the problem here, that the parameters cannot be retrieved? What about instrument other than IRIS/OSIRIS?